t/app/controller/waste_bexley_garden.t

```diff
@@ -889,6 +891,197 @@ FixMyStreet::override_config {
         };
     };
 
+    subtest 'mid-year pro-rata calculation' => sub {
+        set_fixed_time('2024-08-01T00:00:00'); # 6 months after start
+        $mech->delete_problems_for_body($body->id);
+
+        my $uprn = 10001;
+        my $contract_id = 'CONTRACT_456';
+
+        my ($new_sub_report) = $mech->create_problems_for_body(
+            1,
+            $body->id,
+            'Garden Subscription - New',
+            {   category    => 'Garden Subscription',
+                external_id => "Agile-$contract_id",
+                user => $user,
+            },
+        );
+        $new_sub_report->set_extra_fields(
+            { name => 'uprn', value => $uprn } );
+        $new_sub_report->update;
+        FixMyStreet::Script::Reports::send();
+
+        $agile_mock->mock( 'CustomerSearch', sub { {
```

Comment:
I don't know if some of this needs to be folded into the mock stuff that Matthew did. (I can always try that on Monday if need be).

Reply:
Yeah, hopefully not too tricky!